### PR TITLE
fix(pooledMap): return ordered result #2236

### PR DESCRIPTION
**Pull request by Sorikairox**
Based on issue #2236.
Pooled map result are now ordered correctly

### CODE REVIEW

File: async/pool_test.ts [hunk 1]
```diff
@@ -44,3 +44,24 @@ Deno.test("[async] pooledMap errors", async function () {
   });
   assertEquals(mappedNumbers, [3]);
 });
+
+Deno.test("pooledMap returns ordered items", async () => {
```
Comment (kt3k) [inline async/pool_test.ts:48]: nice test!

File: async/pool.ts [hunk 2]
```diff
@@ -40,7 +47,7 @@ export function pooledMap<T, R>(
         // fail the race, taking us to the catch block where all currently
         // executing jobs are allowed to finish and all rejections among them
         // can be reported together.
-        p.then((v) => writer.write(Promise.resolve(v))).catch(() => {});
+        writer.write(p);
```
Comment (kt3k) [inline async/pool.ts:50]: What happens if p rejects with AggregateError?

Comment (kt3k) [inline async/pool.ts:50]: What happens if p rejects with AggregateError?

Comment (Sorikairox) [reply async/pool.ts:50]: Good point.

I think we should handle this with another Exception made specifically for this case, that could not be thrown in p?

However that would be a breaking change.

Maybe check AggregateError message in the condition too?

Comment (Sorikairox) [reply async/pool.ts:52]: @kt3k I improved the handling of `AggregateError` so `controller.error(e as unknown);` is called only when it is relevant.
No interference is `p` rejects with `AggregateError` anymore

Comment (kt3k) [reply async/pool.ts:55]: Thanks for updating! This looks much better 👍

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**